l5kit/l5kit/evaluation/metrics.py

```diff
@@ -1,46 +1,299 @@
     error = -np.log(np.sum(np.exp(error - max_value), axis=-1)) - max_value  # reduce modes
     return error
+
+
+def rmse(ground_truth: np.ndarray, pred: np.ndarray, confidences: np.ndarray, avails: np.ndarray) -> np.ndarray:
+    """
+    Return the root mean squared error, computed using the stable nll
+
+    Args:
+        ground_truth (np.ndarray): array of shape (timesteps)x(2D coords)
+        pred (np.ndarray): array of shape (modes)x(timesteps)x(2D coords)
+        confidences (np.ndarray): array of shape (modes) with a confidence for each mode in each sample
+        avails (np.ndarray): array of shape (timesteps) with the availability for each gt timesteps
+
+    Returns:
+        np.ndarray: negative log-likelihood for this example, a single float number
+
+    """
+    nll = neg_multi_log_likelihood(ground_truth, pred, confidences, avails)
+    _, future_len, _ = pred.shape
+
+    return np.sqrt(2 * nll / future_len)
+
+
+def prob_true_mode(
+    ground_truth: np.ndarray, pred: np.ndarray, confidences: np.ndarray, avails: np.ndarray
+) -> np.ndarray:
+    """
+    Return the probability of the true mode
+
+    Args:
+        ground_truth (np.ndarray): array of shape (timesteps)x(2D coords)
+        pred (np.ndarray): array of shape (modes)x(timesteps)x(2D coords)
+        confidences (np.ndarray): array of shape (modes) with a confidence for each mode in each sample
+        avails (np.ndarray): array of shape (timesteps) with the availability for each gt timesteps
+
+    Returns:
+        np.ndarray: a (modes) numpy array
+
+    """
+    _assert_shapes(ground_truth, pred, confidences, avails)
+
+    ground_truth = np.expand_dims(ground_truth, 0)  # add modes
+    avails = avails[np.newaxis, :, np.newaxis]  # add modes and cords
+
+    error = np.sum(((ground_truth - pred) * avails) ** 2, axis=-1)  # reduce coords and use availability
+
+    with np.errstate(divide="ignore"):  # when confidence is 0 log goes to -inf, but we're fine with it
+        error = np.log(confidences) - 0.5 * np.sum(error, axis=-1)  # reduce timesteps
+
+    # use max aggregator on modes for numerical stability
+    max_value = error.max()  # error are negative at this point, so max() gives the minimum one
+
+    error = np.exp(error - max_value) / np.sum(np.exp(error - max_value))
+    return error
+
+
+def time_displace(
+    ground_truth: np.ndarray, pred: np.ndarray, confidences: np.ndarray, avails: np.ndarray
+) -> np.ndarray:
+    """
+    Return the displacement at timesteps T
+
+    Args:
+        ground_truth (np.ndarray): array of shape (timesteps)x(2D coords)
+        pred (np.ndarray): array of shape (modes)x(timesteps)x(2D coords)
+        confidences (np.ndarray): array of shape (modes) with a confidence for each mode in each sample
+        avails (np.ndarray): array of shape (timesteps) with the availability for each gt timesteps
+
+    Returns:
+        np.ndarray: a (timesteps) numpy array
```

Comment:
What is the dimension of this array? It might be good to include, similarly to the input arrays.